### PR TITLE
Include dhall-nix in stack and cabal project configs

### PR DESCRIPTION
**Pull request by sjakobi**

### CODE REVIEW

**Comment by sjakobi** [comment]
There's a warning:

```
src/Dhall/Nix.hs:211:5: warning: [-Wincomplete-patterns]
    Pattern match(es) are non-exhaustive
    In an equation for ‘loop’: Patterns not matched: (ToMap _ _)
    |
211 |     loop (Const _) = return (Fix (NSet []))
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```

How about we use `-Werror` in CI?

**Comment by sjakobi** [comment]
> How about we use `-Werror` in CI?

Actually the added visibility during development from including `dhall-nix` in the configs should go most of the way…

**Comment by sjakobi** [comment]
So `hnix` depends on `unix`, which doesn't build on windows – I'll revert the changes to `stack-lts-12.yaml` then.

**Comment by Gabriel439** [comment]
@sjakobi: If you add `dhall-nix` to this list then it will enable `-Werror` in CI: https://github.com/dhall-lang/dhall-haskell/blob/master/nix/shared.nix#L149-L154

File: dhall-nix/src/Dhall/Nix.hs [hunk 1]
```diff
@@ -497,6 +497,17 @@ dhallToNix e = loop (Dhall.Core.normalize e)
...
+        a' <- loop a
+        let ks = Fix (NBinary NApp "builtins.attrNames" "kvs")
+        let v = Fix (NBinary NApp (Fix (NBinary NApp "builtins.getAttr" "k")) "kvs")
+        let setBindings =
+                [ NamedVar [StaticKey "mapKey"] "k" Nix.nullPos
+                , NamedVar [StaticKey "mapValue"] v Nix.nullPos
+                ]
+        let map_ = Fix (NBinary NApp "map" (Fix (NAbs "k" (Fix (NSet setBindings)))))
+        let toMap = Fix (NAbs "kvs" (Fix (NBinary NApp map_ ks)))
+        return (Fix (NBinary NApp toMap a'))
```
Comment (sjakobi) [inline dhall-nix/src/Dhall/Nix.hs]: @Gabriel439 Does this look sensible?

    \(x : { a : Bool }) -> toMap x

is translated as

    x:
      (kvs:
        map (k:
          {
            mapKey = k;
            mapValue = builtins.getAttr k kvs;
            }) (builtins.attrNames kvs)) x

Comment (Gabriel439) [reply dhall-nix/src/Dhall/Nix.hs]: @sjakobi: Yeah, that looks correct to me

**Comment by jneira** [comment]
@sjakobi Late to the party, but the project can't be built in windows with `cabal v2-build all` or `stack build` cause `hnix` needs `unix`.
The ci passed in appveyor cause it is using `stack-lts12.yaml` that has not the changes of this pr (i am gonna fix it cause compilations times in windows with `ghc-8.6.5` are acceptable)

**Comment by sjakobi** [comment]
Thanks for looking into this @jneira! I think it would be nice to include `dhall-nix` in the stack build at least on non-Windows systems.

Not sure if configuring this in the cabal file would help:

```
    if os(windows)
        buildable: False
```

**Comment by jneira** [comment]
@sjakobi yeah, i forgot that it makes tools ignore it in windows, thanks!
I will prepare a pr with this fix and making appveyor use the default stack.yaml (upgrading it to ghc-8.6.5)

**Pull request merged**